util/topsql/stmtstats/stmtstats.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stmtstats
+
+import (
+	"sync"
+
+	"go.uber.org/atomic"
+)
+
+// StatementStats is a counter used locally in each session.
+// We can use StatementStats to count data such as "the number of SQL executions",
+// and it is expected that these statistics will eventually be collected and merged
+// in the background.
+type StatementStats struct {
+	mu     sync.Mutex
+	data   StatementStatsMap
+	closed *atomic.Bool
+}
+
+// CreateStatementStats try to create and register an StatementStats.
+// If we are in the initialization phase and have not yet called SetupTopSQL
+// to initialize the top-sql, nothing will happen, and we will get nil. But
+// this scene should never appear, because we always call SetupTopSQL before
+// starting the server, at this moment we cannot receive connections and will
+// not create a valid session. So this case will never happen: "This function
+// returns nil, so this valid session will count nothing".
+func CreateStatementStats() *StatementStats {
+	if manager == nil {
```

Comment:
`write access` only occurs when calling `SetupTopSQL` and `CloseTopSQL`. At present, `read-write access` will never happen. But in order to avoid data race due to changes in the future, it is necessary to use an atomic protection. I'll make update.